extensions/sql-migration/src/api/utils.ts

```diff
@@ -703,6 +705,205 @@ export async function getManagedInstancesDropdownValues(managedInstances: azureR
 	return managedInstancesValues;
 }
 
+// retrieve the auth keys
+export async function retrieveAuthKeys(migrationStateModel: MigrationStateModel): Promise<SqlMigrationServiceAuthenticationKeys> {
+
+	let defaultKeys: SqlMigrationServiceAuthenticationKeys = {
+		authKey1: '',
+		authKey2: ''
+	};
+
+	const service = migrationStateModel._sqlMigrationService;
+	if (service) {
+		const account = migrationStateModel._azureAccount;
+		const subscription = migrationStateModel._sqlMigrationServiceSubscription;
+		const resourceGroup = migrationStateModel._sqlMigrationServiceResourceGroup.name;
+		const location = service.location;
+		const serviceName = service.name;
+		if (service?.properties?.integrationRuntimeState) {
+			service.properties.integrationRuntimeState = undefined;
+		}
+		const keys = await getSqlMigrationServiceAuthKeys(
+			account,
+			subscription,
+			resourceGroup,
+			location,
+			serviceName);
```

Comment:
try add try catch block around async call. It helps handling and logging error conditions.